workspaces/src/network/account.rs

```diff
@@ -40,7 +101,191 @@ impl Contract {
         &self.account.id
     }
 
+    pub fn as_account(&self) -> &Account {
+        &self.account
+    }
+
     pub(crate) fn signer(&self) -> &InMemorySigner {
         self.account.signer()
     }
+
+    /// Call the current contract's function using the contract's own account
+    /// details to do the signing. Returns a Builder object that we will make
+    /// use to populate the rest of the call details.
+    ///
+    /// If we want to make use of the contract's account to call into a
+    /// different contract besides the current one, use
+    /// `contract.as_account().call` instead.
+    pub fn call<'a, T: Network>(
+        &self,
+        worker: &'a Worker<T>,
+        function: String,
+    ) -> CallBuilder<'a, T> {
+        self.account.call(worker, self.id().clone(), function)
+    }
+
+    /// Call a view function into the current contract. Returns a result that
+    /// yields a JSON string object.
+    pub async fn view<T: Network>(
+        &self,
+        worker: &Worker<T>,
+        function: String,
+        args: Vec<u8>,
+    ) -> anyhow::Result<ViewResultDetails> {
+        worker.view(self.id().clone(), function, args).await
+    }
+
+    /// Deletes the current contract, and returns the execution details of this
+    /// transaction. The beneciary will receive the funds of the account deleted
+    pub async fn delete_contract<T: Network>(
+        self,
+        worker: &Worker<T>,
+        beneficiary_id: AccountId,
+    ) -> anyhow::Result<CallExecutionDetails> {
+        self.account.delete_account(worker, beneficiary_id).await
+    }
+}
+
+pub struct CallBuilder<'a, T> {
+    worker: &'a Worker<T>,
+    signer: InMemorySigner,
+    contract_id: AccountId,
+
+    function: String,
+    args: Vec<u8>,
+    deposit: Option<u128>,
+    gas: Option<u64>,
+}
+
+impl<'a, T: Network> CallBuilder<'a, T> {
+    fn new(
+        worker: &'a Worker<T>,
+        contract_id: AccountId,
+        signer: InMemorySigner,
+        function: String,
+    ) -> Self {
+        Self {
+            worker,
+            signer,
+            contract_id,
+            function,
+            args: serde_json::json!({}).to_string().into_bytes(),
+            deposit: None,
+            gas: None,
+        }
+    }
+
+    pub fn with_args(mut self, args: Vec<u8>) -> Self {
+        self.args = args;
+        self
+    }
+
+    pub fn with_args_json<U: serde::Serialize>(mut self, args: U) -> Self {
+        self.args = serde_json::to_vec(&args).unwrap();
+        self
+    }
+
+    pub fn with_args_borsh<U: borsh::BorshSerialize>(mut self, args: U) -> Self {
+        self.args = args.try_to_vec().unwrap();
+        self
+    }
+
+    pub fn with_deposit(mut self, deposit: u128) -> Self {
+        self.deposit = Some(deposit);
+        self
+    }
+
+    pub fn with_gas(mut self, gas: u64) -> Self {
+        self.gas = Some(gas);
+        self
+    }
+
+    pub async fn transact(self) -> anyhow::Result<CallExecutionDetails> {
```

Comment:
Somewhat awkward name, not sure what would be better `.call` perhaps?

Reply:
I think the rationale here, and just my assumption, is just that `transact` makes it more clear that it's signing and sending a transaction rather than just calling a function in a view context, but this might be intuitive for a developer anyway.